### PR TITLE
Texture compression comparison

### PR DESCRIPTION
**Pull request by gpx1000**
## Description

This sample demonstrates how to use different types of compressed GPU textures in a Vulkan application, and shows 
the timing benefits of each.

## Tested Environments

Windows 10 / Ubuntu Linux
GeForce 3060 / GeForce 2080 / GeForce 1070
Vulkan version: 1.2.189
Sample Assets changes
depends upon compressed textures found in KhronosGroup/Vulkan-Samples-Assets#15

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [x] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

### CODE REVIEW

**Comment by SaschaWillems** [comment]
Works fine for me on my desktop GPU.

Two remarks though:
- Switching between target formats, I can't see any difference in the rendering output. Judging by the code, it looks like it's only benchmarking transcoding to a different format. In my opinion it would make sense to use the transcoded textures for the displayed scene. Not sure if that's easily possible with the framework.
- I'd like to see some documentation. There are now comments in the code and the sample doesn't include a readme. Having a few lines of text about what the sample does would be appreciated :)

**Review by gary-sweet — CHANGES_REQUESTED** [review]
This isn't working for me. It tries to use formats that are unsupported.

available_texture_formats is calculated by get_available_texture_formats(), but doesn't appear to actually be used anywhere.

**Comment by gpx1000** [comment]
@gary-sweet please give the latest another shot to see if it's working for you now.

**Comment by gary-sweet** [comment]
> @gary-sweet please give the latest another shot to see if it's working for you now.

@gpx1000 Yep, that's much better thanks.

Can I suggest adding "vkb::StatIndex::gpu_ext_read_bytes" to the requested stats for this sample?
It very obviously then shows a reduction in read bandwidth when choosing a compressed texture, which helps to explain the small performance improvement.

**Comment by gpx1000** [comment]
I added vkb::StatIndex::gpu_ext_read_bytes to the stats requested.  I don't have a device that supports them however, so @gary-sweet will need to request validation that this is all that was required and that it works for what you want displayed here.

**Review by gary-sweet — DISMISSED** [review]
LGTM, thanks.

**Review by SaschaWillems — DISMISSED** [review]
LGTM. All my points have been addressed 👍

File: samples/CMakeLists.txt [hunk 1]
```diff
@@ -89,6 +89,7 @@ set(ORDER_LIST
     "16bit_storage_input_output"
     "16bit_arithmetic"
     "async_compute"
+    "texture_cmpression_comparison"
```
Comment (TomAtkinsonArm) [inline samples/CMakeLists.txt]: `texture_compression_comparison`

**Review by TomAtkinsonArm — COMMENTED** [review]
LGTM

File: samples/CMakeLists.txt (hunk 1 above)
Comment (gpx1000) [reply samples/CMakeLists.txt]: thanks!

**Review by SaschaWillems — APPROVED** [review]

**Review by TomAtkinsonArm — APPROVED** [review]

**Review by gary-sweet — APPROVED** [review]

**Comment by marty-johnson59** [comment]
Merging with 3 approvals  LMK if issues..

**Pull request merged**